index.js

```diff
@@ -314,3 +305,30 @@ module.exports = fp(
         fastify: '3.x',
     }
 );
+
+function onRouteBind(metrics) {
+    // map id to route url
+    const idToRoute = new Map();
+    return function (options) {
+        options.config = options.config || {};
+        options.config.metrics = options.config.metrics || {};
+        options.config.metrics.routeId = getRouteId(options.config);
+
+        // Check for duplicated routes
+        if (idToRoute.has(getRouteId(options.config))) {
+            throw new Error(
+                `Duplicated id, you have just used the id "${getRouteId(
```

Comment:
Here I would chagnge the message into something like: `Duplicated routeId ${routeId}`

Reply:
Maintaining the rest of the message?